### ISSUE TITLE
Look into using `LambdaMetaFactory` to limit catching `Throwable`s

### ISSUE THREAD

**Issue by jcksndfrd** [jcksndfrd/azure-sdk-for-java#47]

**Comment by jcksndfrd** [comment]
[Link](https://github.com/Azure/azure-sdk-for-java/pull/31325#discussion_r992597299) to PR discussion

**Comment by 20004427** [comment]
https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/LambdaMetafactory.html

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/ReflectionSerializable.java#L167

**Comment by ajpetherick** [comment]
**The Problem**
The azure-json-reflect library contains 73 instances of `Throwable` across 9 classes. We catch `Throwables` so much as, according to the [java docs](https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/MethodHandle.html),  `MethodHandle.invokeExact()` and `MethodHandle.invoke()` are declared to throw Throwable hence we must as well.
`public final Object invoke(Object... args)
                    throws Throwable`
Best practices state that catching throwables should be used sparingly as `Throwable` is the superclass of all errors and exceptions. Errors such as `StackOverflowError` are thrown by the Java Virtual Machine(JVM), and are unable to be handled by the library. 
**The Solution**
As [suggested](https://github.com/Azure/azure-sdk-for-java/pull/31325#discussion_r992597299) by Alan Zimmer, `LambdaMetaFactory` can be used to turn the call sites into function types.  This means that any direct `MethodHandle` can be converted into an implementation of the desired functional interface, and the mandatory `Throwable` can be consolidated into one static method such as [createMetaFactory()](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/ReflectionSerializable.java#L167) 
**Scope**
Integrating `LambdaMetaFactory` into the azure-json-reflect library is a large change set that requires a good amount of refactoring better suited to a PR of its own in a future patch release of the library.

**Comment by jcksndfrd** [comment]
I have successfully implemented the use of `LambdaMetaFactory` in `GsonJsonReader` in the branch `metafactory`. There is now only one caught `Throwable`.

Once this is done for all `JsonReader`s and `JsonWriter`s it will be merged into `feature/azure-json-reflect` and hence the PR.

**Comment by ajpetherick** [comment]
Nice work @jcksndfrd, I have a partially implemented `JacksonJsonWriter` which I will press on with and commit to the `metafactory` branch.

**Comment by jcksndfrd** [comment]
@ajpetherick also added it to `GsonJsonWriter`
I have refactored the initialize method to a `static` block so that the JVM can optimize it
